charmcraft/commands/build.py

```diff
@@ -434,7 +427,10 @@ def pack_charm_in_instance(
         if self.shell_after:
             cmd.append("--shell-after")
 
-        emit.progress(f"Launching environment to pack for base {build_on}")
+        emit.progress(
+            f"Launching environment to pack for base {build_on} "
+            "(may take a while the first time but it's reusable)"
```

Comment:
I think the environment launch time in charmcraft may appear longer than what it actually is because the actual charm build process is silent. It's possible that if more progress is printed during the charm build, it may seem faster.

Note that this will require the outer terminal control to be paused so that the inner instance can print messages (see issue #582)

Reply:
Ack, thanks!